src/main/java/org/truffleruby/Main.java

```diff
@@ -143,11 +143,11 @@ public static void main(String[] args) {
                     if (config.shouldUsePathScript()) {
                         context.setOriginalInputFile(config.getScriptFileName());
                         //language=ruby
-                        bootCode = "Kernel.exit Truffle::Boot.main_s";
+                        bootCode = "Truffle::Boot.main_s";
```

Comment:
Only the one for `main_s`. But these `exit` were really just a heavy way to pass the return value to the Main.